tasks/main.yml

```diff
@@ -39,27 +42,38 @@
   command: setcap cap_net_bind_service=+ep /usr/bin/caddy
   when: caddy_bind_cap.rc > 0
 
-- name: Create directories
-  file: path={{ item }} state=directory owner=caddy
-  with_items:
-    - /etc/caddy
-    - /var/www
-
 - name: Create Caddyfile
   copy: content="{{caddy_config}}" dest=/etc/caddy/Caddyfile
   notify: Restart Caddy
 
-- name: Upstart service
-  template: src=caddy.conf dest=/etc/init/caddy.conf
+- name: Config for Upstart service from github
+  get_url: url=https://raw.githubusercontent.com/mholt/caddy/master/dist/init/linux-upstart/caddy.conf dest=/etc/init/caddy.conf mode=0755
   ignore_errors: yes
   when: not caddy_init_system or caddy_init_system == "upstart"
   notify: Restart Caddy
 
-- name: Systemd service
-  template: src=caddy.service dest=/etc/systemd/system/caddy.service
+- name: Systemd service | get config from github
+  get_url: url=https://raw.githubusercontent.com/mholt/caddy/master/dist/init/linux-systemd/caddy.service dest=/etc/systemd/system/caddy.service mode=0755
   ignore_errors: yes
   when: not caddy_init_system or caddy_init_system == "systemd"
   notify: Reload systemd service file
 
+- name: Systemd service | set ExecStart
+  lineinfile: 
+    state: present
+    dest: /etc/systemd/system/caddy.service
+    regexp: '^ExecStart\='
+    line: 'ExecStart=/usr/bin/caddy -log stdout -agree=true -email {{ caddy_email }} -conf=/etc/caddy/Caddyfile -root=/var/www'
+  when: not caddy_init_system or caddy_init_system == "systemd"
+  notify: Reload systemd service file
+
+- name: Systemd service | fix port permission error
+  lineinfile: 
+    state: absent
+    dest: /etc/systemd/system/caddy.service
+    regexp: '^NoNewPrivileges\=true'
```

Comment:
I doubt this is needed. I guess computerlyrik experienced a bug with xenolf/lego, has worked around said bug, and this happened to be an artifact of his prior testing. At least that's my speculation based on having tried to reproduce the original report on Gentoo and Debian Jessie.